src/NuGet.Clients/NuGet.PackageManagement.UI/Xamls/PackageManagerControl.xaml.cs

```diff
@@ -44,6 +44,14 @@ public partial class PackageManagerControl : UserControl, IVsWindowSearch
         // list in response to PackageSourcesChanged event.
         private bool _dontStartNewSearch;
 
+        // When executing a UI operation, we disable the PM UI and ignore any refresh requests.
+        // This tells the operation execution part that it needs to trigger a refresh when done.
+
+        private bool _uiRefreshRequired;
+
+        // Signifies where an action is being executed. Should be updated in a coordinated fashion with IsEnabled
+        private bool _executingAction;
```

Comment:
`_isExecutingAction`

Based on your comment above, can you just use `IsEnabled` instead?

Reply:
IsEnabled is a UIElement property can only be invoked on the UI thread. 
We shouldn't need to switch to the UI thread to do this check.